weave/trace_server/trace_server_interface.py

```diff
@@ -212,6 +221,26 @@ class TableSchemaForInsert(BaseModel):
     rows: list[dict[str, Any]]
 
 
+class OtelExportReq(BaseModel):
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+    project_id: str
+    traces: ExportTraceServiceRequest
+    wb_user_id: Optional[str] = Field(None, description=WB_USER_ID_DESCRIPTION)
+
+
+# Spec requires that the response be of type Export<signal>ServiceResponse
+# https://opentelemetry.io/docs/specs/otlp/
+class OtelExportRes(BaseModel):
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+    _response: ExportTraceServiceResponse
+
+    def __init__(self, **data: Any) -> None:
+        super().__init__(**data)
+        # Set the response type to ExportTraceServiceResponse
+        # TODO: Actually populate the error fields
```

Comment:
The error field population might actually not be super important here. I think it's to communicate partial successes so that the client knows which span exports it should retry. If an error is raised the full thing should retry. Since we do this batched all in one go, if an error is raised and it wants to retry it should do everything anyways I think.

Since in the end we communicate back with JSONResponse instead of protobuf message I'm not sure the client even knows how to correctly parse this.

Other implementations just always return a newly initialized trace service response at the end of execution regardless so this is probably fine as is, but wanted to leave a comment in case it comes up later.